### PR TITLE
multiclient: add retry option and sequentially retry each eth client

### PR DESCRIPTION
**Pull request by tailingchen**

### CODE REVIEW

File: multiclient/option.go [hunk 1]
```diff
@@ -33,3 +35,35 @@ func EthURLs(urls []string) Option {
...
+	// Timeout is the timeout for each retry. Set to 0 means use default timeout 5 seconds.
+	Timeout time.Duration
+	// Delay is the delay duration for each retry. Set to 0 means use default delay 1 second.
+	Delay time.Duration
+}
+
+// WithRetryConfig configures the parameters for request retry.
+func WithRetryConfig(retry RetryConfig) Option {
+	return func(mc *Client) error {
+		log.Info("Use given retry config", "retryLimit", retry.Limit, "retryDelay", retry.Delay)
```
Comment (markya0616) [inline multiclient/option.go]: print timeout?

File: multiclient/client.go [hunk 2]
```diff
@@ -572,40 +533,25 @@ func (mc *Client) CallContext(ctx context.Context, isPostToAll bool, result inte
...
-	}
 
-	if err != nil {
-		log.Debug("Failed to perform batch JSON-RPC calls on any eth client", "err", err)
-		return err
+	var errs []error
+
+	finalErr := mc.requestRetryFunc(ctx, clients, func(ctx context.Context, rpcClient *rpc.Client) (bool, error) {
+		var err error
+		err = rpcClient.BatchCallContext(ctx, b)
```
Comment (markya0616) [inline multiclient/client.go]:
```
var err error
err = rpcClient.BatchCallContext(ctx, b)
```
can be rewritten into
```
err := rpcClient.BatchCallContext(ctx, b)
```

**Review by markya0616 — APPROVED** [review]

**Review by markya0616 — APPROVED** [review]

File: multiclient/option.go (hunk 1 above)
Comment (tailingchen) [reply multiclient/option.go]: fixed

File: multiclient/client.go (hunk 2 above)
Comment (tailingchen) [reply multiclient/client.go]: fixed

**Pull request merged**